14172 - Update "How does it work?"

ketaki-deodhar opened the pull request:
*Issue #:*
https://github.com/bcgov/entity/issues/14172

*Description of changes:*
Update "How does it work?" text

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

kialj876 commented:
Do these changes need to wrapped in a feature flag? I don't think we would want to release this until the different company types are available in the business app.

It would probably be beneficial to have the text come from the feature flag unless all the new types are getting launched at the same time

ketaki-deodhar commented:
> Do these changes need to be wrapped in a feature flag? I don't think we would want to release this until the different company types are available in the business app.
> 
> You might need different variations of this if the types aren't being released at the same time

Yes. You are right. Missed it. Will fix it.

ketaki-deodhar commented:
> Do these changes need to be wrapped in a feature flag? I don't think we would want to release this until the different company types are available in the business app.
> 
> You might need different variations of this if the types aren't being released at the same time

Added feature flag and fixed tests.

severinbeauvais commented:
Please update app version. Re: https://github.com/bcgov/sbc-auth/blob/main/auth-web/package.json

seeker25 commented:
Also: Are these flags setup in DEV/TEST/PROD?

severinbeauvais commented:
> Also: Are these flags setup in DEV/TEST/PROD?

Yes, this is an existing flag in LD sbc-auth project.
Dev: true
Test: true
Production: false

severinbeauvais reviewed (approved):
LGTM.

Please create a new Tech Debt ticket to remove this flag and conditional code once this feature is deployed to Prod.

kialj876 commented on auth-web/src/views/auth/home/IncorpOrRegisterView.vue:
```diff
@@ -14,41 +14,80 @@
-                  </v-list-item-subtitle>
-                </v-list-item-content>
-              </v-list-item>
+              <div v-if="enableBcCccUlc">
```
can you please add a computed value for `bulletPoints` that changes based on the `enableBcCccUlc` flag instead of having the v-if/else clause?

kialj876 commented on auth-web/src/views/auth/home/IncorpOrRegisterView.vue:
```diff
@@ -14,41 +14,80 @@
-                </v-expansion-panel-content>
-              </v-expansion-panel>
-            </v-expansion-panels>
+            <div v-if="enableBcCccUlc">
```
same thing here for `expansionPanels`

kialj876 commented on auth-web/src/views/auth/home/MaintainBusinessView.vue:
```diff
@@ -5,22 +5,43 @@
-              </v-list-item>
-            </v-list-item-content>
-          </v-list-item>
+          <div v-if="enableBcCccUlc">
```
same idea here

kialj876 commented on auth-web/src/components/auth/common/NumberedCompanyTooltip.vue:
```diff
@@ -1,24 +1,40 @@
   <v-tooltip top max-width="450px" light content-class="tooltip">
     <template v-slot:activator="{ on }">
-      <span v-on="on" class="tooltip-text">Numbered Benefit Company</span>
+      <span v-on="on" class="tooltip-text" v-if="enableBcCccUlc">Numbered Company</span>
```
pls rmv v-if/else and do a computed value for the text here

kialj876 commented on auth-web/src/components/auth/common/NumberedCompanyTooltip.vue:
```diff
@@ -1,24 +1,40 @@
-      <span>A Benefit Company can choose to use as its name the incorporation number of the company followed by “B.C.
-        Ltd.” The incorporation number is assigned by the Business Registry after the Incorporation Application is filed
-        and the company is incorporated.</span>
+      <div v-if="enableBcCccUlc">
```
same thing in here

kialj876 reviewed (changes requested):
Hey Ketaki,

You're PR introduces a lot of duplicate code that can be avoided by using computed values for your text instead of v-if/else blocks, can you please update it based on my comments?
It will make it easier in the future to read / refactor / do styling updates etc.

Other than that I think it is looking good 👍
